### PR TITLE
[Feature Store] Fix `get_online_feature_service` for "Viewer" user usage.

### PR DESCRIPTION
**Pull request by davesh0812**
[ML-4137](https://jira.iguazeng.com/browse/ML-4137)

### CODE REVIEW

File: mlrun/feature_store/retrieval/storey_merger.py [hunk 1]
```diff
@@ -146,27 +145,26 @@ def init_online_vector_service(
...
             driver = get_online_target(featureset)
             if not driver:
                 raise mlrun.errors.MLRunInvalidArgumentError(
                     f"resource {featureset.uri} does not have an online data target"
                 )
             cache.cache_table(featureset.uri, driver.get_table_object())
+
             for key in featureset.spec.entities.keys():
-                if (
-                    not self.vector.spec.with_indexes
```
Comment (assaf758) [inline mlrun/feature_store/retrieval/storey_merger.py:158]: why did you remove this line?

Comment (davesh0812) [reply mlrun/feature_store/retrieval/storey_merger.py:158]: It is not really necessary there because inside the `get` function in line 529 I check it before delete those columns from the response.

**Review by assaf758 — APPROVED** [review]
